queries/weather.py

```diff
@@ -176,16 +184,40 @@ def _round_to_nearest_hour(t):
 _RVK_STATION_ID = 1
 
 
-def _curr_observations(query):
-    """ Fetch latest weather observation data from nearest weather station """
+def _curr_observations(query, result):
+    """ Fetch latest weather observation data from weather station closest
+        to the location associated with the query (i.e. either user location 
+        coordinates or a specific placename) """
     loc = query.location
 
+    # User asked about a specific location
+    # Try to find a matching Icelandic placename
+    if "location" in result and result.location != "Ísland":
+
+        # Some strings should never be interpreted as Icelandic placenames
+        if result.location in ICE_PLACENAME_BLACKLIST:
+            return None
+
+        # Unfortunately, many foreign country names are also Icelandic
+        # placenames, so we automatically exclude country names.
+        cc = isocode_for_country_name(result.location)
```

Comment:
Is this required here?

Reply:
As long as the weather module can't provide information for locations outside Iceland, we don't want queries about foreign places (e.g. "á Spáni", "í París") to return data for Icelandic placenames with the same name.

Reply:
Understood!